langworld_db_data/adders/listed_value_adder.py

```diff
@@ -89,6 +131,57 @@ def _add_to_inventory_of_listed_values(
 
         return id_of_new_value
 
+    @staticmethod
+    def _get_indices_and_their_line_numbers_in_features_listed_values(
+        rows: list[dict[str, str]],
+        feature_id: str,
+        new_value_en: str,
+        new_value_ru: str,
+    ) -> list[dict[str, int]]:
+
+        value_indices_to_inventory_line_numbers: list[dict[str, int]] = []
+        """List of dictionaries, each mapping value index to line number in features_listed_values,
+        ex. [{"index": 1, "line number": 4}, {"index": 2, "line number": 5}, {"index": 3, "line number": 6}].
+        Contains all indices and line numbers of values with given feature_id.
+        """
+
+        for i, row in enumerate(rows):
+            if row["feature_id"] != feature_id:
+                continue
+
+            if row["en"] == new_value_en or row["ru"] == new_value_ru:
+                raise ValueError(f"Row {row} already contains value you are trying to add")
+
+            value_index = int(row["id"].split(ID_SEPARATOR)[-1])
+            value_indices_to_inventory_line_numbers.append(
+                {
+                    "index": value_index,
+                    "line number": i,
+                }
+            )
+
+        return value_indices_to_inventory_line_numbers
+
+    @staticmethod
+    def _increment_ids_whose_indices_are_not_less_than_index_to_assign_in_rows(
+        rows: list[dict[str, str]],
+        value_indices_to_inventory_line_numbers: list[dict[str, int]],
+        index_to_assign: int,
+    ):
+
+        for value_index_and_line_number in value_indices_to_inventory_line_numbers:
+            if value_index_and_line_number["index"] < index_to_assign:
```

Comment:
note that the literals "index" and "line number" are used more than once in the code. If they are used only twice, that's more or less fine. If more, we should definitely create two module-level constants for them.

Some say that even if these literals are used just twice each, creation of constants is already warranted

Reply:
Am I right that, in this case, at the beginning of the document `INDEX` and `LINE_NUMBER` constants should be introduced? I'm not sure how this should be done. Could you give an example, please?

Reply:
yep, but if they are really only used twice each, let's leave it as it is

Reply:
The search shows that each of them is used seven times of which four are in the code itself (and three within dosctrings)

Reply:
5 times in tests don't count. But three times in working code make creation of constant(s) warranted.

Reply:
Done!